### PR TITLE
[PyTorch] Mask R-CNN from Torchvision

### PR DESCRIPTION
**Pull request by dkurt**
https://pytorch.org/vision/master/_modules/torchvision/models/detection/mask_rcnn.html

* For better efficiency, use `box_detections_per_img` option to limit maximum number of detected objects and `rpn_post_nms_top_n_test` to limit RPN's number of proposals.

res

### CODE REVIEW

**Comment by dkurt** [comment]
@likholat, please review

File: modules/mo_pytorch/mo_extensions/front/pytorch/detection_output_ext.py [hunk 1]
```diff
@@ -28,13 +28,22 @@ class DetectionOutputExtractor(FrontExtractorOp):
...
+            'code_type',
+            'share_location',
+            'background_label_id',
+            'clip_before_nms',
+        ]
+
+        attrs = {}
+        for attr in attr_names:
+            if hasattr(node.module, attr):
+                attrs[attr] = getattr(node.module, attr)
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/front/pytorch/detection_output_ext.py:46]: We had a cast to `int` for the `variance_encoded_in_target` attribute before. Don't we need this cast anymore?

Comment (dkurt) [reply modules/mo_pytorch/mo_extensions/front/pytorch/detection_output_ext.py:46]: Yes, seems like we don't need it anymore. At least tested with OpenVINO 2021.4 and master branch.

File: modules/mo_pytorch/mo_extensions/front/pytorch/elementwise_ext.py [hunk 2]
```diff
@@ -23,6 +23,8 @@
 from mo.ops.power import AttributedPower
 from extensions.ops.activation_ops import *
 from mo.ops.const import Const
+from mo.ops.clamp import Clamp
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/front/pytorch/elementwise_ext.py:26]: Unused import

File: modules/mo_pytorch/mo_extensions/front/pytorch/roi_align_ext.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from mo.front.common.partial_infer.utils import int64_array
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/front/pytorch/roi_align_ext.py:19]: Unused import

File: modules/mo_pytorch/mo_extensions/front/pytorch/roi_align_ext.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from mo.front.extractor import FrontExtractorOp
+from extensions.ops.roialign import ROIAlign
+from mo.utils.error import Error
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/front/pytorch/roi_align_ext.py:22]: Unused import

File: modules/mo_pytorch/mo_extensions/front/pytorch/elementwise_ext.py (hunk 2 above)
Comment (dkurt) [reply modules/mo_pytorch/mo_extensions/front/pytorch/elementwise_ext.py]: Done, thanks! We need to enable a static analyzer in future.

File: modules/mo_pytorch/mo_extensions/load/pytorch/hooks.py [hunk 5]
```diff
@@ -138,10 +134,27 @@ def size(self, dim=None):
...
+            torch.int64: np.int64,
+            torch.float32: np.float32,
+        }
+        if not dtype in mapping:
+            raise Exception("Unexpected dst type:", dtype)
+
+        class Cast(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.dst_type = mapping[dtype]
```
Comment (likholat) [inline modules/mo_pytorch/mo_extensions/load/pytorch/hooks.py:155]: Don't we need to pass `dtype` as a constructor parameter?

Comment (dkurt) [reply modules/mo_pytorch/mo_extensions/load/pytorch/hooks.py:155]: Yes, but actually class uses a global `dtype` from method's parameters. Probably that's not clear and better to refactor by a separate PR. Local class definition is also a problem but not a critical one.

**Review by likholat — APPROVED** [review]
